### PR TITLE
FT2-292 display version data

### PR DESCRIPTION
**Pull request by andycyberian**

### CODE REVIEW

File: DFC.ServiceTaxonomy.VersionComparison/Models/Parts/ObjectValue.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        public override string ToString()
+        {
+            if (!string.IsNullOrWhiteSpace(Html))
+            {
+                return Html;
+            }
+            if (!string.IsNullOrWhiteSpace(Text))
+            {
+                if (Text.Equals("true", StringComparison.InvariantCultureIgnoreCase) ||
```
Comment (ravi-methods) [inline DFC.ServiceTaxonomy.VersionComparison/Models/Parts/ObjectValue.cs:19]: Could possibly use bool.TryParse, so we don't have to compare strings twice?

File: DFC.ServiceTaxonomy.VersionComparison/Models/Parts/ObjectValue.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                    Text.Equals("false", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return Text.Equals("true", StringComparison.InvariantCultureIgnoreCase) ? "Yes" : "No";
+                }
+                return Text;
+            }
+            if (!string.IsNullOrWhiteSpace(Value))
+            {
+                if (Value.Equals("true", StringComparison.InvariantCultureIgnoreCase) ||
```
Comment (ravi-methods) [inline DFC.ServiceTaxonomy.VersionComparison/Models/Parts/ObjectValue.cs:28]: Same as above.

File: DFC.ServiceTaxonomy.VersionComparison/Services/DiffBuilderService.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        private List<DiffItem> MergeDictionaries(Dictionary<string, PropertyExtract> basePartDictionary,
+            Dictionary<string, PropertyExtract> comparePartDictionary)
+        {
+            var diffList = new List<DiffItem>();
+            var keys = basePartDictionary.Keys.Union(comparePartDictionary.Keys);
+            foreach (string key in keys)
+            {
+                diffList.Add(new DiffItem{
+                    Name = basePartDictionary.ContainsKey(key) ? basePartDictionary[key].Name : comparePartDictionary[key].Name,
```
Comment (ravi-methods) [inline DFC.ServiceTaxonomy.VersionComparison/Services/DiffBuilderService.cs:77]: 3 lookups for the same key, should extract to a variable perhaps?

File: DFC.ServiceTaxonomy.VersionComparison/Services/DiffBuilderService.cs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                    BaseItem = basePartDictionary.ContainsKey(key) ? basePartDictionary[key].Value : string.Empty,
+                    BaseURLs = basePartDictionary.ContainsKey(key) ? basePartDictionary[key].Links : null,
+                    CompareItem = comparePartDictionary.ContainsKey(key) ? comparePartDictionary[key].Value : string.Empty,
```
Comment (ravi-methods) [inline DFC.ServiceTaxonomy.VersionComparison/Services/DiffBuilderService.cs:80]: 2 lookups here as above.

File: DFC.ServiceTaxonomy.VersionComparison/Models/Parts/ObjectValue.cs (hunk 1 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.VersionComparison/Models/Parts/ObjectValue.cs]: Amended

File: DFC.ServiceTaxonomy.VersionComparison/Services/DiffBuilderService.cs (hunk 3 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.VersionComparison/Services/DiffBuilderService.cs]: Amended

File: DFC.ServiceTaxonomy.VersionComparison/Models/Parts/ObjectValue.cs (hunk 2 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.VersionComparison/Models/Parts/ObjectValue.cs]: Amended

File: DFC.ServiceTaxonomy.VersionComparison/Services/DiffBuilderService.cs (hunk 4 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.VersionComparison/Services/DiffBuilderService.cs]: Amended

**Review by ravi-methods — APPROVED** [review]

**Pull request merged**